x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/JdbcHttpClient.java

```diff
@@ -30,17 +30,23 @@
 class JdbcHttpClient {
     private final HttpClient httpClient;
     private final JdbcConfiguration conCfg;
-    private final InfoResponse serverInfo;
+    private InfoResponse serverInfo;
```

Comment:
It can still be final if there is an `else { serverInfo = null }`  in the ctor.

Reply:
But then [this](https://github.com/elastic/elasticsearch/pull/40641/files/ff6fcfc1116985637ae73b45b677bf9914cab19a..12826537874ae3007e70263ee170f161ab2e0f6c#diff-b2c6208433c105ca91e765533b2c82fcR87) won't work.